homeassistant/components/songpal/media_player.py

```diff
@@ -65,52 +63,49 @@
 )
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Songpal platform."""
-    if PLATFORM not in hass.data:
-        hass.data[PLATFORM] = {}
-
-    if discovery_info is not None:
-        name = discovery_info["name"]
-        endpoint = discovery_info["properties"]["endpoint"]
-        _LOGGER.debug("Got autodiscovered %s - endpoint: %s", name, endpoint)
+async def async_setup_platform(
+    hass: HomeAssistantType, config: dict, async_add_entities, discovery_info=None
+) -> None:
+    """Support legacy configuration file."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+        )
+    )
 
-        device = SongpalDevice(name, endpoint)
-    else:
-        name = config.get(CONF_NAME)
-        endpoint = config.get(CONF_ENDPOINT)
-        device = SongpalDevice(name, endpoint, poll=False)
 
-    if endpoint in hass.data[PLATFORM]:
-        _LOGGER.debug("The endpoint exists already, skipping setup.")
-        return
+async def async_setup_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up songpal media player."""
+    name = config_entry.data[CONF_NAME]
+    endpoint = config_entry.data[CONF_ENDPOINT]
+    model = config_entry.data.get(CONF_MODEL)
 
+    device = SongpalDevice(name, endpoint, model)
     try:
         await device.initialize()
     except SongpalException as ex:
         _LOGGER.error("Unable to get methods from songpal: %s", ex)
         raise PlatformNotReady
 
-    hass.data[PLATFORM][endpoint] = device
-
     async_add_entities([device], True)
 
+    platform = entity_platform.current_platform.get()
```

Comment:
Has this change something to do with config flow? If not, it would be better to create a separate PR for it :-)